BoxContentSDK/BoxContentSDK/BOXURLSessionCacheClient.m

```diff
@@ -0,0 +1,297 @@
+{
+    NSString *path = [self dirPathOfSessionTaskWithBackgroundSessionId:backgroundSessionId sessionTaskId:sessionTaskId];
+    return [self createDirectory:path error:error];
+}
+
+- (BOOL)createDirectory:(NSString *)path error:(NSError **)error
+{
+    BOOL isDir = NO;
+    BOOL success = YES;
+    if ([[NSFileManager defaultManager] fileExistsAtPath:path isDirectory:&isDir] == NO || isDir == NO) {
+        success = [[NSFileManager defaultManager] createDirectoryAtPath:path withIntermediateDirectories:YES attributes:nil error:error];
+    }
+    return success;
+}
+
+- (BOOL)deleteDirectory:(NSString *)directoryPath error:(NSError **)error
+{
+    BOOL isDir = NO;
+    BOOL success = YES;
+    if ([[NSFileManager defaultManager] fileExistsAtPath:directoryPath isDirectory:&isDir] == YES && isDir == YES) {
+        success = [[NSFileManager defaultManager] removeItemAtPath:directoryPath error:error];
+    }
+    return success;
+}
+
+- (BOOL)createFile:(NSString *)path error:(NSError **)error
+{
+    BOOL success = [[NSFileManager defaultManager] createFileAtPath:path contents:nil attributes:nil];
+    if (success == NO && error != nil) {
+        (*error) = [[NSError alloc] initWithDomain:BOXContentSDKErrorDomain code:BOXContentSDKSCacheErrorFileCreateFailed userInfo:nil];
+    }
+    return success;
+}
+
+// path sessions/$backgroundSessionId/$sessionTaskId
+- (NSString *)dirPathOfSessionTaskWithBackgroundSessionId:(NSString *)backgroundSessionId sessionTaskId:(NSUInteger)sessionTaskId
+{
+    return [[[self.cacheDir stringByAppendingPathComponent:BOXURLSessionTaskCacheSessionsDirName] stringByAppendingPathComponent:backgroundSessionId] stringByAppendingPathComponent:[NSString stringWithFormat:@"%lu", (unsigned long)sessionTaskId]];
+}
+
+// path users/$userId/$associateId/$backgroundSessionId-$sessionTaskId
+- (NSString *)dirPathOfUserSessionTaskGivenUserId:(NSString *)userId associateId:(NSString *)associateId
+{
+    return [[[self.cacheDir stringByAppendingPathComponent:BOXURLSessionTaskCacheUsersDirName] stringByAppendingPathComponent:userId] stringByAppendingPathComponent:associateId];
+}
+
+- (NSString *)filePathOfUserSessionTaskGivenUserId:(NSString *)userId associateId:(NSString *)associateId backgroundSessionId:(NSString *)backgroundSessionId sessionTaskId:(NSUInteger)sessionTaskId
+{
+    NSString *path = [self dirPathOfUserSessionTaskGivenUserId:userId associateId:associateId];
+    NSString *fileName = [self fileNameGivenBackgroundSessionId:backgroundSessionId sessionTaskId:sessionTaskId];
+    return [path stringByAppendingPathComponent:fileName];
+}
+
+- (NSString *)fileNameGivenBackgroundSessionId:(NSString *)backgroundSessionId sessionTaskId:(NSUInteger)sessionTaskId
+{
+    return [NSString stringWithFormat:@"%@-%lu", backgroundSessionId, (unsigned long)sessionTaskId];
+}
+
+- (BOXURLBackgroundSessionIdAndSessionTaskId *)parseBackgroundSessionIdAndSessionTaskIdFileName:(NSString *)name
```

Comment:
Minor: this method name is hard to read. Can we simplify this to something else?